src/base/bittorrent/session.cpp

```diff
@@ -2160,6 +2134,27 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
         }
     }
 
+    // Common
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    p.flags &= ~lt::add_torrent_params::flag_duplicate_is_error; // Already checked
+#else
+    p.flags &= ~lt::torrent_flags::duplicate_is_error; // Already checked
+#endif
+    // Make sure the torrent will be initially checked and then paused
+    // to perform some service jobs on it. We will start it if needed.
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    p.flags |= lt::add_torrent_params::flag_paused;
+    p.flags |= lt::add_torrent_params::flag_auto_managed;
+    p.flags |= lt::add_torrent_params::flag_stop_when_ready;
+#else
+    p.flags |= lt::torrent_flags::paused;
+    p.flags |= lt::torrent_flags::auto_managed;
+    p.flags |= lt::torrent_flags::stop_when_ready;
+#endif
+
+    p.max_connections = maxConnectionsPerTorrent();
```

Comment:
Why don't you allow to restore these two from resume data?

Reply:
IMO, it is safer to always enforce our limits. (edge case:: the conf file has been edited manually)

Reply:
The question is "isn't possible to change these limits for particular torrent from qBittorrent?".

Reply:
No. We expose only the ratio limits per torrent to the user. Changes to the max connections/uploads are done in bulk for all torrents.